rust/src/fvm/engine.rs

```diff
@@ -318,12 +322,26 @@ mod v2 {
                         .into_iter()
                         .filter_map(|tr| match tr {
                             ExecutionEvent2::GasCharge(charge) => {
+                                // We set the gas for "negative" charges to 0. This isn't correct,
+                                // but it won't matter in practice (the sum of the gas charges in
+                                // the trace aren't guaranteed to equal the total gas charge
+                                // anyways).
```

Comment:
@arajasek has said that this is fine. We only use tracing for benchmarking/debugging gas anyways.